pkg/kubelet/status/manager.go

```diff
@@ -382,6 +382,11 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+			if pod.DeletionGracePeriodSeconds != nil && *pod.DeletionGracePeriodSeconds == 0 {
```

Comment:
The only way to fix that is conditional delete.  The design of
graceful made Kubelets responsible for pod lifecycle, and I'm
incredibly wary of changing that right now.  I'd prefer to look for
fixed that minimize the impact of the race condition.